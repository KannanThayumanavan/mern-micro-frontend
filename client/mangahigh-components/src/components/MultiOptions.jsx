import React from 'react';
import PropTypes from 'prop-types';

const MultiOptions = ({ 
  options, groupName, getSelectedValue,
}) => options.map((option) => (
  <div 
    key={`${groupName}_${option}`} 
    className="form-check form-check-inline p-3" 
    onChange={(e) => getSelectedValue(e.target.value)}
  >
    <input className="form-check-input" type="radio" name={groupName} value={option} id={option} />
    <label className="form-check-label" htmlFor={option}>{option}</label>
  </div>
));

MultiOptions.propTypes = {
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
  groupName: PropTypes.string.isRequired,
  getSelectedValue: PropTypes.func.isRequired,
};
      
export default MultiOptions;