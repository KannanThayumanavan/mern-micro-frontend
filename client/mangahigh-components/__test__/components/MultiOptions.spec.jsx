import React from 'react';
import { shallow } from 'enzyme';
import MultiOptions from '../../src/components/MultiOptions';

describe('Test MultiOptions Component', () => {  
	it('with required props', () => {
    const multiOptionsValues = ['Option 1', 'Option 2'];
		const renderedModule = shallow(
     <MultiOptions 
      options={multiOptionsValues} 
      groupName='test_multi_option' 
      getSelectedValue={jest.fn()} 
    />);
    expect(renderedModule).toMatchSnapshot();
	});

  it('should return selected value on change', () => {
    const multiOptionsValues = ['Option 1'];
    const mockCallback = jest.fn();
    const eventObj = {
      target: {
        value: 'test',
      },
    };
    const renderedModule = shallow(<MultiOptions 
      options={multiOptionsValues} 
      groupName='test_multi_option' 
      getSelectedValue={mockCallback} 
    />);    
    renderedModule.simulate('change', eventObj);
    expect(mockCallback).toHaveBeenCalledWith(eventObj.target.value);
  });
});