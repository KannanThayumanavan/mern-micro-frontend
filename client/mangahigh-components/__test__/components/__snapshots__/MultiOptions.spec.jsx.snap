// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MultiOptions Component with required props 1`] = `
Array [
  <div
    className="form-check form-check-inline p-3"
    key="test_multi_option_Option 1"
    onChange={[Function]}
  >
    <input
      className="form-check-input"
      id="Option 1"
      name="test_multi_option"
      type="radio"
      value="Option 1"
    />
    <label
      className="form-check-label"
      htmlFor="Option 1"
    >
      Option 1
    </label>
  </div>,
  <div
    className="form-check form-check-inline p-3"
    key="test_multi_option_Option 2"
    onChange={[Function]}
  >
    <input
      className="form-check-input"
      id="Option 2"
      name="test_multi_option"
      type="radio"
      value="Option 2"
    />
    <label
      className="form-check-label"
      htmlFor="Option 2"
    >
      Option 2
    </label>
  </div>,
]
`;
